### ISSUE TITLE
[$500] Chat - Page blinks when message new user via profile screen

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#31002]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.96-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Go to FAB> Start chat
3. Enter new e-mail and create chat
4. Tag this user in the compose box
5. Click on tagged user e-mail
6. From the opened profile page Message this user

## Expected Result:
Chat window should open smoothly, no blinking and errors should appear

## Actual Result:
Page blinks when message new user via profile screen

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/4b8936a8-8372-4aa4-a661-be7f7006556b

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01279fb989323f7c3e</li>
        <li>Upwork Job ID: 1721907582953017344</li>
        <li>Last Price Increase: 2023-11-07</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Contributor | 27671912</li>
</ul></li>
    </ul>
</details>

**Comment by NisargDeveloper** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The page is blinking twice. 

### What is the root cause of that problem?

It seems the function/useEffect is running twice. Once we pass empty array in the useEffect. It should run fine.

### What changes do you think we should make in order to solve the problem?

We gotta add useEffect. And we can pass the empty array to resolve this issue.

```
function ExampleComponent() {
  useEffect(() => {
    // This code will run once when the component mounts

    // Perform any initialization or setup tasks here

    return () => {
      // This code will run when the component unmounts (cleanup)
      // You can return a cleanup function here if needed
    };
  }, []); // Empty dependency array means this effect runs only once

  return (
    // JSX for your component
  );
}
```


### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by sophiepintoraetz** [comment]
waiting on @burczu to review proposals!

**Comment by sophiepintoraetz** [comment]
Still on @burczu for a review!

**Comment by burczu** [comment]
@sophiepintoraetz We closed the PR from @dukenv0307 as turned out this solution won't work. So now, we are still waiting for proposals here.

**Comment by burczu** [comment]
Hey @sophiepintoraetz! I'll be OOO till January 6th, and right after that I'm leaving the C+ role and joining Waves. Could you assign another C+ engineer to handle this issue from now on? Thanks.

**Comment by situchan** [comment]
I can take over if needed. Seems like we're waiting proposals

**Comment by twisterdotcom** [comment]
Hello new issue. Still a very real thing:

https://github.com/Expensify/App/assets/9133401/c1e24761-d601-45b8-b44b-432c3cc79fd0

**Comment by twisterdotcom** [comment]
Okay, do we want to stick "Help Wanted" back on this, or does this require a back end change? @puneetlath and @cead22 - I see we said this in closing the original PR: https://github.com/Expensify/App/pull/31425#issuecomment-1839033469

>I don't think we should change the way mentions work for this PR. We have some planned changes to the "hidden" behavior, so I think we should just tackle those separately.

Do we want to close this? Mark it `Internal` and put it on a wave? Or make it External again and get a better proposal?

**Comment by cead22** [comment]
Thanks for the summary @twisterdotcom, I added the help wanted label because the root cause still seems accurate and we just need new proposals. 

Ideally, the new proposal should make it so that looking for personal details works for every case, and the code to do that is reused everywhere we look for personal details in the code

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by twisterdotcom** [comment]
Okay, I also can't reproduce this. Closing now.
